server/web_ui/admin/src/components/admin_objectgraph.rs

```diff
@@ -0,0 +1,294 @@
+                        ObjectType::BuiltinGroup
+                    } else {
+                        ObjectType::Group
+                    }
+                } else if classes.contains(&"account".to_string()) {
+                    if classes.contains(&"person".to_string()) {
+                        ObjectType::Person
+                    } else {
+                        ObjectType::ServiceAccount
+                    }
+                } else {
+                    return None;
+                };
+
+                // Filter out the things we want to keep, if the filter is empty we assume we want all.
+                if !filters.contains(&obj_type) && !filters.is_empty() {
+                    return None;
+                }
+
+                let spn = entry.attrs.get("spn")?.first()?;
+                Some((spn.clone(), uuid.clone(), obj_type))
+            }).collect::<HashSet<(String, String, ObjectType)>>();
+
+
+        // Vec<obj, uuid, obj's members>
+        let members_of = entries.into_iter().filter_map(|entry| {
+            let spn = entry.attrs.get("spn")?.first()?.clone();
+            let uuid = entry.attrs.get("uuid")?.first()?.clone();
+            let keep = typed_entries.iter().any(|(_, filtered_uuid, _)| { &uuid == filtered_uuid });
+            if keep {
+                Some((spn, uuid, entry.attrs.get("member")?.clone()))
+            } else {
+                None
+            }
+        }).collect::<Vec<_>>();
+
+        // Constructing graph source
+        let mut sb = String::new();
+        sb.push_str("digraph {\n  rankdir=\"RL\"\n");
+        for (spn, _, members) in members_of {
+            members.iter()
+                .filter(|member| typed_entries.iter().any(|(spn, _, _)| spn == *member))
+                .for_each(|member| {
+                    sb.push_str(format!(r#"  "{spn}" -> "{member}"{}"#, "\n").as_str());
+                });
+        }
+
+        for (spn, uuid, obj_type) in typed_entries {
+            let (color, shape, route) = match obj_type {
+                ObjectType::Group => ("#b86367", "box", AdminRoute::ViewGroup { uuid }),
+                ObjectType::BuiltinGroup => ("#8bc1d6", "component", AdminRoute::ViewGroup { uuid }),
+                ObjectType::ServiceAccount => ("#77c98d", "parallelogram", AdminRoute::ViewServiceAccount { uuid }),
+                ObjectType::Person => ("#af8bd6", "ellipse", AdminRoute::ViewPerson { uuid }),
+            };
+            let url = route.to_path();
+            sb.push_str(format!(r#"  "{spn}" [color = "{color}", shape = {shape}, URL = "{url}"]{}"#, "\n").as_str());
+        }
+        sb.push_str("}");
+        init_graphviz(&sb.as_str());
+
+        let on_checkbox_click = {
+            let scope = ctx.link().clone();
+            move |_: Event| {
+                let coll = document().get_elements_by_class_name("obj-graph-filter-cb");
+                let mut filters = vec![];
+
+                for i in 0..*&coll.length() {
```

Comment:
This feels weird, especially with the unchecked_into later... is there a reason it's not just `for option in coll.iter()` or similar?

Reply:
You can't iterate on HtmlCollection afaik

Reply:
I went with a different approach now